### ISSUE TITLE
A11y Refactor: `MeetupList` Component

### ISSUE THREAD

**Issue by TylerAPfledderer** [ethereum/ethereum-org-website#8580]
## Objective
Improving accessibility and UI markup for the [MeetupList](https://github.com/ethereum/ethereum-org-website/blob/dev/src/components/MeetupList.tsx) component, including UX with screen readers.

## Description
Now that this component has been migrated to ChakraUI for the UI Epic, there might be further improvements that can be made.

- As this is a list, it could be rendered with Chakra's `List` components and possibly an `aria-label` to clarify what the list is, else a screen reader would not declare to the user what group of items they are navigating into (cognitive load consideration).
- The search input should include at least hidden text for the screen reader to state that "results will update as you type". This text can be in a separate element connected to the input via `aria-describedby` (maybe the preferred way)
- If a user makes a search in the input that turns up no results, there should be a live region surrounding the info banner that immediately displays to have the screen reader assertively read out it's message.
- Should there be search results, because the list is in a `List` component as suggested above, the user will be given the number of items that rendered once they navigate into the list.

**Comment by corwintines** [comment]
This sounds great @TylerAPfledderer. Is this something you would be interested in taking on?

**Comment by TylerAPfledderer** [comment]
@corwintines Yes sir!

**Comment by corwintines** [comment]
Wicked! Ill assign to you :)
